### ISSUE TITLE
[Bug Report] API Compatibility and GymEnvironment wrapper

### ISSUE THREAD

**Issue by Rohan138** [Farama-Foundation/Gymnasium#302]
### Describe the bug

It seems that the ```GymEnvironment``` environment and the API compatibility wrapper are applied in the wrong order for environments that are registered with gym **and** use the old API.

Use case: I'm working on migrating [mbrl-lib](https://github.com/facebookresearch/mbrl-lib) to ```gymnasium```. However, ```mbrl-lib``` currently supports environments from [pybullet-gym](https://github.com/benelot/pybullet-gym) which still uses ```gym.registry```, and use the old step API.

I'm trying to use ```env = gymnasium.make("GymEmvironment-v0", env_id="HumanoidPyBulletEnv-v0", apply_api_compatibility=True)``` to convert the pybullet-gym env into a gymnasium env using the new API.

However, ```env.reset()``` fails for me (see stack trace). It looks like this is because the ```EnvCompatibility``` wrapper is applied before the ```GymEnvironment``` wrapper, which means the ```GymEnvironment``` wrapper still expects a gym environment that uses the new API, which is not what I intended. How can I pass the ```apply_api_compatibility=True``` down to ```gym``` instead?

### Code example

```shell
>>> import gymnasium as gym
>>> import pybulletgym
>>> env = gym.make("GymV26Environment-v0", env_id="HumanoidPyBulletEnv-v0", apply_api_compatibility=True)
pybullet build time: Jan 30 2023 14:22:52
/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gym/spaces/box.py:127: UserWarning: WARN: Box bound precision lowered by casting to float32
  logger.warn(f"Box bound precision lowered by casting to {self.dtype}")
WalkerBase::__init__
>>> env.reset()
/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gym/utils/passive_env_checker.py:174: UserWarning: WARN: Future gym versions will require that `Env.reset` can be passed a `seed` instead of using `Env.seed` for resetting the environment random number generator.
  logger.warn(
/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gym/utils/passive_env_checker.py:187: UserWarning: WARN: Future gym versions will require that `Env.reset` can be passed `options` to allow the environment initialisation to be passed additional information.
  logger.warn(
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gymnasium/wrappers/order_enforcing.py", line 42, in reset
    return self.env.reset(**kwargs)
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gymnasium/wrappers/env_checker.py", line 45, in reset
    return env_reset_passive_checker(self.env, **kwargs)
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gymnasium/utils/passive_env_checker.py", line 192, in env_reset_passive_checker
    result = env.reset(**kwargs)
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gymnasium/wrappers/compatibility.py", line 96, in reset
    return self.env.reset(), {}
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gymnasium/envs/external/gym_env.py", line 67, in reset
    return self.gym_env.reset(seed=seed, options=options)
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gym/wrappers/time_limit.py", line 68, in reset
    return self.env.reset(**kwargs)
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gym/wrappers/order_enforcing.py", line 42, in reset
    return self.env.reset(**kwargs)
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gym/wrappers/env_checker.py", line 45, in reset
    return env_reset_passive_checker(self.env, **kwargs)
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gym/utils/passive_env_checker.py", line 192, in env_reset_passive_checker
    result = env.reset(**kwargs)
TypeError: WalkerBaseBulletEnv.reset() got an unexpected keyword argument 'seed'
```

### System info

Gymnasium 0.26.3
gym 0.26.3
MacOS, Python 3.10.8

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/Gymnasium/issues) in the repo

**Comment by Rohan138** [comment]
Closed, nvm: ```env = gym.make("HumanoidPyBulletEnv-v0", apply_api_compatibility=True)```, then ```env = gymnasium.make("GymV26Environment-v0", env=env)``` works as intended. My apologies!

**Comment by pseudo-rnd-thoughts** [comment]
Are you able to use `gymnasium.make("GymV22Environment-v0", env_id="HumanoidPyBulletEnv-v0")` if you have `gym` and `shimmy` installed?

**Comment by Rohan138** [comment]
By the way, I can't use ```gymnasium=0.27.1```, had to downgrade to 0.26.3, I get an error related(?) to shimmy I think.

```
>>> import gymnasium as gym
>>> import pybulletgym
>>> gym.make("GymV22Environment-v0", env_id="HumanoidPyBulletEnv-v0")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gymnasium/envs/registration.py", line 654, in make
    raise e
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gymnasium/envs/registration.py", line 642, in make
    env = env_creator(**_kwargs)
TypeError: _raise_shimmy_error() got an unexpected keyword argument 'env_id'
>>> gym.make("GymV26Environment-v0", env_id="HumanoidPyBulletEnv-v0")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gymnasium/envs/registration.py", line 654, in make
    raise e
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gymnasium/envs/registration.py", line 642, in make
    env = env_creator(**_kwargs)
TypeError: _raise_shimmy_error() got an unexpected keyword argument 'env_id'

```

**Comment by Rohan138** [comment]
With ```gymnasium==0.26.3```:
```
>>> import gymnasium as gym
>>> import pybulletgym
>>> gym.make("GymV22Environment-v0", env_id="HumanoidPyBulletEnv-v0")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gymnasium/envs/registration.py", line 569, in make
    _check_version_exists(ns, name, version)
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gymnasium/envs/registration.py", line 219, in _check_version_exists
    _check_name_exists(ns, name)
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gymnasium/envs/registration.py", line 197, in _check_name_exists
    raise error.NameNotFound(
gymnasium.error.NameNotFound: Environment GymV22Environment doesn't exist. Did you mean: `GymV26Environment`?
>>> 
```

**Comment by Rohan138** [comment]
That works too.

**Comment by Rohan138** [comment]
Actually, sorry, this works with env.reset(), but not env.step() somehow (the first solution does):
```
>>> env.step(env.action_space.sample())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gymnasium/wrappers/order_enforcing.py", line 38, in step
    return self.env.step(action)
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gymnasium/wrappers/env_checker.py", line 37, in step
    return env_step_passive_checker(self.env, action)
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gymnasium/utils/passive_env_checker.py", line 230, in env_step_passive_checker
    result = env.step(action)
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/shimmy/openai_gym_compatibility.py", line 242, in step
    obs, reward, done, info = self.gym_env.step(action)
  File "/Users/rapotdar/miniforge3/envs/mllib/lib/python3.10/site-packages/gym/wrappers/time_limit.py", line 50, in step
    observation, reward, terminated, truncated, info = self.env.step(action)
ValueError: not enough values to unpack (expected 5, got 4)
```

**Comment by pseudo-rnd-thoughts** [comment]
Are you using Gym v0.26? I think there is an issue when you are using a gym v0.21 environment with gym v0.26 installed and shimmy. 
Is it fixed if you use gym v0.21 rather than v0.26?
